### ISSUE TITLE
`#13354 diagnose some incorrect usages of the question mark operator` creates incorrect error

### ISSUE THREAD

**Issue by Alexandcoats** [rust-lang/rust-analyzer#13429]
The changes made by #13354 cause the following incorrect error in this minimal example using `axum`:

```rust
use axum::{routing::get, Router, Server};

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    Server::bind(&([0, 0, 0, 0], 8080).into())
        .serve(
            Router::new()
                .route("test", get(|| async { "hello" }))
                .into_make_service(),
        )
        .await?;

    Ok(())
}
```

```
the trait `Try` is not implemented for `_`rust-analyzer[not-implemented](https://rust-analyzer.github.io/manual.html#not-implemented)
```

**rust-analyzer version**: `0.3.1248-standalone (0531aab52 2022-10-16)`

**rustc version**: `rustc 1.66.0-nightly (a37499ae6 2022-09-18)`

**Comment by flodiebold** [comment]
Hmm yeah this is the general "preventing follow-up errors" problem. I don't think you can really tell from the type that there was a problem, since these placeholders can also occur legitimately, but looking for placeholders and error types might be the best approach for now.
